code/game/machinery/kitchen/microwave.dm

```diff
@@ -435,16 +435,20 @@
 	return ffuu
 
 /obj/machinery/microwave/CtrlClick(mob/user)
-    if(!user.incapacitated() && Adjacent(user) && user.dexterity_check() && anchored)
-        cook() //Cook checks for power, brokenness, and contents internally
-        return
-    return ..()
+	if(!user.incapacitated() && Adjacent(user) && user.dexterity_check() && anchored)
+		if(issilicon(user) && !attack_ai(user))
```

Comment:
If you wanted to be *really* OOP, you could make it a `/machinery`-level list and add calls to a module-checking proc in all of the base-level machinery access procs like `/machinery/attack_ai()` and such. You'd also need to make sure the `AltClick()`, `CtrlClick()`, et cetera of all machines made a check against their parent to make sure the borg can access it.

That'd lay a good framework for being able to restrict any machine you want to any module or set of modules.